docs/specs/mlnet-database-loader/mlnet-database-loader-specs.md

```diff
@@ -0,0 +1,463 @@
+
+    See [related issue](https://github.com/dotnet/machinelearning-samples/issues/507)
+
+- **Support for remote/network connection and internal connection within the RDBMS server**: The Database loader should not only support remote/network connection to the RDBMS server but also support from C# running within the RDBMS server such as [SQL Server CLR integration](https://docs.microsoft.com/en-us/sql/relational-databases/clr-integration/clr-integration-overview?view=sql-server-2017), [Oracle Database Extensions for .NET](https://docs.oracle.com/cd/B19306_01/win.102/b14306/intro.htm), etc.  Scenarios:
+
+    1. Training-machine-ML.NET-code <--network--> Remote database-in-RDBMS-server
+
+    2. RDBMS-server running database and .NET code using ML.NET code
+
+- **Implementation as a separate NuGet package**: Since the implementation of this feature uses the primitives in System.Data, therefore this feature should be available as a separate NuGet package so that ML.NET core packages don't depend on System.Data.
+
+- **Support for sparse data**: The database loader should support sparse data, at least up to the maximum number of columns in SQL Server (1,024 columns per nonwide table, 30,000 columns per wide table or 4,096 columns per SELECT statement).
+
+    ML.NET supports sparse data such as in the following example using a [sparse matrix](https://en.wikipedia.org/wiki/Sparse_matrix) of thousands or even millions of columns even when in this example only 200 columns have real data (sparse data): 
+
+    - [ML.NET sample using millions of columns with sparse data](https://github.com/dotnet/machinelearning-samples/tree/master/samples/csharp/getting-started/LargeDatasets)
+
+    SQL Server supports [Sparse columns](https://docs.microsoft.com/en-us/sql/relational-databases/tables/use-sparse-columns?view=sql-server-2017), however, it is just a way to optimize storage for null values. It still needs to have a real column created in the table per each logical column (i.e. 1,000 columns defined in the SQL table) even when it might not have data.
+
+
+
+
+
+
+# Gated conditions for first Preview release
+
+- TBD
+
+# Gated conditions for GA release
+
+- TBD
+
+# API design examples
+
+The following are a few C# examples on how this API could be used by a developer when loading data from a database into an IDataView.
+
+Whenever in the samples below it is using a `SqlConnection` type, it could also be using any other supported DbConnection in .NET providers, such as:
+
+- `SqlConnection` (SQL Server)
+- `NpgsqlConnection` (PostgreSQL)
+- `OracleConnection` (Oracle)
+- `MySqlConnection` (MySql)
+- etc.
+
+The specific type (`SqlConnection`, `NpgsqlConnection`, `OracleConnection`, etc.) is specified as a generic parameter so references to the needed NuGet packages will be provided by the user when trying to compile.
```

Comment:
Those types would be specified in the user's code, not in ML.NET code. Our parameters are Interfaces or base classes.
We prefer the users to need to provide a type (such as `SqlConnection`, `NpgsqlConnection`, `OracleConnection`) in their code so at that time they'll need to provide the dependency to their specific chosen database provider, that's the only reason we're specifying to provide a DbConnection type instead of a string with the provider such as "SQLServer" but then they'll need to know what NuGet assembly to add, etc. IF not provided properly they'll get an execution exception. Instead, by providing a type (such as `NpgsqlConnection`) it won't compile until you add the needed NuGet package.